c/tskit/genotypes.c

```diff
@@ -623,7 +617,75 @@ tsk_vargen_update_site(tsk_vargen_t *self)
          * not-missing */
         num_missing -= (tsk_size_t) no_longer_missing;
     }
-    var->has_missing_data = num_missing > 0;
+    self->has_missing_data = num_missing > 0;
+out:
+    return ret;
+}
+
+int
+tsk_tree_get_variant(tsk_tree_t *self, const tsk_site_t *site, tsk_variant_t *variant,
+    tsk_flags_t options)
+{
+    int ret = 0;
+    tsk_size_t num_samples_alloc;
+
+    variant->tree = self;
```

Comment:
There's too much internal state manipulation of the variant/vargen going on here - why isn't the allocation being done in ``variant_init``? We're doing O(n) operations and allocs here which we should be avoiding if possible.